src/pages/Doeun/Main/Components/Article/Article.js

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from 'react';
+import CommentBar from '../CommentBar/CommentBar';
+import UnderBar from '../UnderBar/UnderBar';
+import Comment from '../Comment/Comment';
+import './Article.scss';
+
+class Article extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      commentList: [],
+      commentValue: '',
+    };
+  }
+
+  componentDidMount() {
+    fetch('http://localhost:3000/data/Doeun/commentData.json', {
+      method: 'GET',
+    })
+      .then(res => res.json())
+      .then(data => {
+        this.setState({
+          commentList: data[this.props.index],
+        });
+      });
+  }
+
+  addComment = () => {
+    let userId = document.location.search.slice(
+      4,
+      document.location.search.indexOf('&')
+    );
+    userId = userId ? userId : 'unknown';
+    this.setState({
+      commentList: this.state.commentList.concat({
+        id: userId,
+        comment: this.state.commentValue,
+      }),
+    });
+  };
+
+  setInput = comment => {
+    this.setState({ commentValue: comment });
+  };
+
+  setComment = comment => {
+    this.setState({ commentList: comment });
+  };
```

Comment:
여기도 계산된 속성을 사용할 수 있을것같아요 !

Reply:
계산된 속성은 어떤 걸 말씀하시는 건가요?!

Reply:
저도 체크리스트에서 본건데 
```suggestion
 setComment = comment => {
   this.setState({ [e.target.name] : comment })
   };
   
   ...
   <Comment name="commentList" ... />
   <CommentBar name="commentValue" .../>
   ...
```
저렇게 [e.target.name] 변수명으로 접근할 수 있더라구요!
그런데 컴포넌트 내부의 엘리먼트를 클릭하는 거라 쪼꼼 복잡해지는 느낌도 있지만,,
이 상황에서도 이게 더 효율적인지 ,, 모루겟네요 😭

어쨌든 중복된 함수 줄이는 데에 요론 방법도 있습니다 !!
notion://www.notion.so/wecode/onChange-33fb203a41fd4d48bdba133b61751d4a